ros2_batch_job/__main__.py

```diff
@@ -274,6 +277,7 @@ def process_coverage(args, job):
 
 def build_and_test(args, job):
     coverage = args.coverage and args.os == 'linux'
+    compile_on_clang = args.compile_on_clang and args.os == 'linux'
```

Comment:
Why is this specific to linux?

Reply:
Currently we are only setting the cmake args for compiler paths to clang paths based on this flag. Given that os x clang compiler is already working as expected, I want to avoid adding unnecessary cmake args or any such logic to this workflow.